src/clj/puppetlabs/services/file_serving/file_metadata/file_metadata_core.clj

```diff
@@ -0,0 +1,200 @@
+(ns puppetlabs.services.file-serving.file-metadata.file-metadata-core
+  (:import (org.apache.commons.codec.digest DigestUtils)
+           [java.text SimpleDateFormat]
+           [java.util Locale Date])
+  (:require [puppetlabs.puppetserver.ringutils :as ringutils]
+            [puppetlabs.puppetserver.liberator-utils :as utils]
+            [puppetlabs.services.protocols.puppet-fileserver-config :as fileserver-config]
+            [clojure.tools.logging :as log]
+            [clojure.java.io :as io]
+            [clojure.string :as string]
+            [schema.core :as schema]
+            [cheshire.core :as cheshire]
+            [compojure.core :as compojure :refer [GET ANY PUT]]
+            [liberator.core :refer [defresource]]
+            [liberator.representation :as representation]
+            [liberator.dev :as liberator-dev]
+            [puppetlabs.services.file-serving.config.puppet-fileserver-config-core :as config]
+            [puppetlabs.services.file-serving.utils.posix-utils :as posix-utils]
+            )
+  (:use [clojure.java.io :only (input-stream)]
+        [ring.middleware.params]
+        ))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; 'handler' functions for HTTP endpoints
+
+(defn- checksum-of-file
```

Comment:
Might be preferable to use something like https://github.com/tebeka/clj-digest instead of rolling our own here.